ecs-agent/api/appnet/client.go

```diff
@@ -18,12 +18,22 @@ import (
 	"fmt"
 	"net"
 	"net/http"
+
+	prometheus "github.com/prometheus/client_model/go"
 )
 
 type appnetClientCtxKey int
 
-type client struct {
+// AppnetClient is an interface with customized Appnet client that
+// implements the GetStats and DrainInboundConnections
+type AppNetClient interface {
+	GetStats(adminSocketPath string, statsRequest string) (map[string]*prometheus.MetricFamily, error)
+	DrainInboundConnections(adminSocketPath string, drainRequest string) error
+}
+
+type AppNetAgentClient struct {
```

Comment:
Nit: `Client` would have been fine (just like `http.Client` below). since this is already scoped to the `appnet` package, there won't be ambiguities. If we compare `appnet.Client` and `appnet.AppNetAgentClient`, the latter seems a bit redundant.

Reply:
Thank you for bringing it up. Actually you are right, it would have been okay to leave it as client. I brought in the change for parity with the other agent.